### ISSUE TITLE
Dynamic class example, outlier does not render 

### ISSUE THREAD

**Issue by dorisjlee** [zenvisage/zenvisage#142]
In real estate tutorial dataset example, if you create the class : forclosureratio : [min,5],[5,max]. The outlier doesn't render when dynamic_class is selected as Category axis.

![image](https://user-images.githubusercontent.com/5554675/29630605-6241ea9e-8802-11e7-8951-1884e0597f81.png)
Error message on the frontend, no backend errors: 
```
angular.min.js:117 TypeError: Cannot read property 'xData' of undefined
    at displayRepresentativeResultsHelper (helper.js:530)
    at Object.displayRepresentativeResults (app.js:398)
    at app.js:1127
    at angular.min.js:96
    at angular.min.js:130
    at n.$eval (angular.min.js:144)
    at n.$digest (angular.min.js:142)
    at n.$apply (angular.min.js:145)
    at l (angular.min.js:97)
    at H (angular.min.js:101)
```

**Comment by adityagp** [comment]
@dorisjlee will try this again

**Comment by dorisjlee** [comment]
@superchaoran  I tried this again by testing it on the server: had to first add the extra line in the schema file as described in [#143](https://github.com/zenvisage/zenvisage/issues/143) and used the above example (forclosureratio : [min,5],[5,max])  then switched to dynamic class view, outlier still doesn't show with the same error.

**Comment by adityagp** [comment]
@dorisjlee will look into the frontend

**Comment by siddta** [comment]
There are only two visualizations for the issued query, both of which make the representative trends. Since a visualization can't be representative and outlier at the same time, the backend returns an empty result for outliers -- that makes sense to me.
To fix this, the front-end should check whether the returned result is empty; if it is then it should not show anything for outliers. @dorisjlee.

**Comment by dorisjlee** [comment]
There was something in the backend that was enforcing the same number of outliers and representative trends which was breaking this. I've fixed this and tested it in the case where I set filter as city='Kendall', which also had the same issue. The dynamic class example above cannot be temporarily tested because dynamic_class is not displayed on the frontend (it seems like the schema-free file upload still did not resolve [Issue #143](https://github.com/zenvisage/zenvisage/issues/143) ) @superchaoran might want to look into this.
